lib/shopwareStoreCommander.js

```diff
@@ -0,0 +1,364 @@
+
+            if (review.status.id !== 1) {
+                // Review finished
+                return review;
+            } else if (pollCount === 20) {
+                // Max polls reached
+                throw new Error(
+                    'Reviews is taking longer than expected. Please check the review status online at ' +
+                    'https://account.shopware.de/'
+                );
+            }
+
+            // Poll review again
+            return pollReviewStatus(review);
+        };
+
+        this.logger.emit('info', `Requesting review of plugin ${plugin.name}...`);
+
+        const res = await this.client.post(`plugins/${plugin.id}/reviews`);
+        // Save the review
+        const review = res.data;
+        plugin.reviews.push(review);
+
+        // Wait for the review to finish
+        await pollReviewStatus(review);
+
+        return plugin;
+    }
+
+    /**
+     * @param {Object} plugin
+     * @return {Promise}
+     */
+    async enablePartialIonCubeEncryption(plugin) {
+        // Check the plugin for the 'encryptionIonCube' addon
+        const encryptionAddon = plugin.addons.find(addon => addon.name === 'encryptionIonCube');
+        if (!encryptionAddon) {
+            // The plugin is not encrpyted, hence don't enable partial encrpytion either
+            return plugin;
+        }
+
+        // Check the plugin for the 'partialIonCubeEncryptionAllowed' addon
+        const partialEncryptionAddonName = 'partialIonCubeEncryptionAllowed';
+        let partialEncryptionAddon = plugin.addons.find(addon => addon.name === partialEncryptionAddonName);
+        if (partialEncryptionAddon) {
+            this.logger.emit('info', `Partial ionCube encryption for plugin ${plugin.name} already enabled`);
+
+            return plugin;
+        }
+
+        // Find and add the addon for partial ionCube encryption
+        partialEncryptionAddon = this.statics.addons.find(addon => addon.name === partialEncryptionAddonName);
+        if (!partialEncryptionAddon) {
+            throw new Error(
+                `Cannot enable partial ionCube encryption for plugin ${plugin.name} due to missing plugin addon option`
+            );
+        }
+        plugin.addons.push(partialEncryptionAddon);
+
+        this.logger.emit('info', `Enabling partial ionCube encryption for plugin ${plugin.name}...`);
+
+        const res = await this.client.put(`plugins/${plugin.id}`, plugin);
+        // Save the updated data locally
+        this.updatePluginData(this.accountData.plugins[plugin.name], res.data);
+
+        return this.accountData.plugins[plugin.name];
+    }
+
+    /**
+     * @param {Object} plugin
+     * @param {Array} fields
+     * @return {Promise}
+     */
+    loadExtraPluginFields(plugin, fields) {
+        plugin.scsLoadedExtraFields = plugin.scsLoadedExtraFields || [];
+        // Load all extra fields
+        const extraFieldPromises = fields.map(field => this.client.get(plugin[field]).then((res) => {
```

Comment:
Have you tried using an `async` closure here as the iterator? If I'm not mistaken, that should simply return a `Promise` too.